system/Session/Handlers/RedisHandler.php

```diff
@@ -102,20 +103,21 @@ protected function setSavePath(): void
             throw SessionException::forEmptySavepath();
         }
 
-        if (preg_match('#(?:tcp://)?([^:?]+)(?:\:(\d+))?(\?.+)?#', $this->savePath, $matches)) {
-            if (! isset($matches[3])) {
-                $matches[3] = ''; // Just to avoid undefined index notices below
+        if (preg_match('#(?:(^.*)://)?([^:?]+)(?:\:(\d+))?(\?.+)?#', $this->savePath, $matches)) {
```

Comment:
probably better to use possible values, eg: `(tcp|tls)`, that's we can have less possible regression.

Reply:
Done